src/Mvc/Mvc.Core/src/Controllers/IControllerActivatorProvider.cs

```diff
@@ -23,19 +23,19 @@ public interface IControllerActivatorProvider
         /// </summary>
         /// <param name="descriptor">The <see cref="ControllerActionDescriptor"/>.</param>
         /// <returns>The delegate used to dispose the activated controller.</returns>
-        Action<ControllerContext, object> CreateReleaser(ControllerActionDescriptor descriptor);
+        Action<ControllerContext, object>? CreateReleaser(ControllerActionDescriptor descriptor);
 
         /// <summary>
         /// Creates an <see cref="Action"/> that releases a controller.
         /// </summary>
         /// <param name="descriptor">The <see cref="ControllerActionDescriptor"/>.</param>
         /// <returns>The delegate used to dispose the activated controller.</returns>
-        Func<ControllerContext, object, ValueTask> CreateAsyncReleaser(ControllerActionDescriptor descriptor)
+        Func<ControllerContext, object, ValueTask>? CreateAsyncReleaser(ControllerActionDescriptor descriptor)
         {
             var releaser = CreateReleaser(descriptor);
             return (context, controller) =>
             {
-                releaser(context, controller);
+                releaser?.Invoke(context, controller);
```

Comment:
If releaser is null then the func returned could be a constant noop func that avoids this closure.